File: BE/src/main/java/org/example/features/order/entity/OrderStatus.java
package org.example.features.order.entity;

/**
 * Order Status Enum
 * Represents the lifecycle of a customer order
 */
public enum OrderStatus {
    /**
     * Order imported from Excel, waiting for admin approval before quote review
     */
    PENDING_APPROVAL,

    /**
     * Order created, waiting for admin to provide quote
     */
    PENDING_QUOTE,

    /**
     * Quote provided, waiting for customer payment
     */
    AWAITING_PAYMENT,

    /**
     * Deposit (60%) received, order is confirmed, pending full processing
     * Đã cọc 60%, đang chờ xử lý
     */
    DEPOSITED,

    /**
     * Payment received, order is being processed
     */
    PROCESSING,

    /**
     * Payment received for READY_MADE order; awaiting delivery preparation
     */
    AWAITING_DELIVERY,

    /**
     * Order handed to shipping carrier / delivery unit
     */
    SHIPPING,

    /**
     * 7 days before delivery_date for CUSTOM_MANUFACTURING:
     * customer must pay the remaining 30%
     */
    AWAITING_REMAINING_PAYMENT,

    /**
     * Order completed and delivered
     */
    COMPLETED,

    /**
     * Order cancelled by admin or customer
     */
    CANCELLED
}

File: FE/src/constants/orderStatus.js
export const UNKNOWN_STATUS_LABEL = 'Không xác định'

export const ORDER_STATUS_LABELS = Object.freeze({
  PENDING_APPROVAL: 'Chờ duyệt đơn',
  PENDING_QUOTE: 'Chờ báo giá',
  AWAITING_CONTRACT: 'Chờ xác nhận hợp đồng',
  AWAITING_PAYMENT: 'Chờ thanh toán',
  DEPOSITED: 'Đã cọc',
  PROCESSING: 'Đang gia công',
  AWAITING_REMAINING_PAYMENT: 'Chờ thanh toán đợt 2',
  AWAITING_DELIVERY: 'Chờ giao hàng',
  SHIPPING: 'Đang giao hàng',
  COMPLETED: 'Hoàn thành',
  CANCELLED: 'Đã hủy',
})

export const REVIEW_STATUS_LABELS = Object.freeze({
  PENDING_REVIEW: 'Chờ đánh giá',
  APPROVED: 'Đã duyệt',
  REJECTED: 'Từ chối',
  NEED_DISCUSSION: 'Cần trao đổi',
})

export const getOrderStatusLabel = (status, overrides = null) => {
  if (!status) return UNKNOWN_STATUS_LABEL

  if (overrides && overrides[status]) {
    return overrides[status]
  }

  return ORDER_STATUS_LABELS[status] || UNKNOWN_STATUS_LABEL
}

export const getReviewStatusLabel = (status) => {
  if (!status) return UNKNOWN_STATUS_LABEL
  return REVIEW_STATUS_LABELS[status] || UNKNOWN_STATUS_LABEL
}
